platform_plugin_saleor/saleor_client/utils.py

```diff
@@ -0,0 +1,208 @@
+    'IntegerField': 'NUMERIC',
+    'DecimalField': 'NUMERIC',
+    'FloatField': 'NUMERIC',
+}
+
+
+def generate_saleor_product_attribute_data(
+    model_cls, model_field: str, product_attrb_name: str
+) -> dict:
+    """
+    Generate the data structure for a Saleor product attribute based on a model field.
+
+    Args:
+        model_cls: The Django model class.
+        model_field: The name of the field in the model.
+        product_attrb_name: The desired name for the Saleor product attribute.
+
+    Returns:
+        dict: A dictionary representing the Saleor attribute data.
+    """
+    model_field_type = get_model_field_type(model_cls, model_field)
+    input_type = ATTRIBUTE_TYPES_MAP.get(model_field_type, "PLAIN_TEXT")
+
+    return {
+        'name': product_attrb_name,
+        'externalReference': model_field,
+        'type': 'PRODUCT_TYPE',
+        'inputType': input_type,
+    }
+
+
+def get_model_field_type(model_cls, field_name: str) -> str:
+    """
+    Get the internal type name of a field in a Django model.
+
+    Args:
+        model_cls: The Django model class.
+        field_name: The name of the field.
+
+    Returns:
+        str: The internal type name of the field (e.g., 'CharField', 'DateTimeField').
+
+    Raises:
+        ValueError: If the field does not exist in the model.
+    """
+    try:
+        field = model_cls._meta.get_field(field_name)
+        return field.get_internal_type()
+
+    except AttributeError as exc:
+        raise ValueError(
+            f"Field '{field_name}' does not exist in model '{model_cls.__name__}'."
+        ) from exc
+
+
+def clean_edges_and_nodes(response: dict) -> list[dict]:
+    """
+    Clean the edges and nodes from the GraphQL response.
+
+    Args:
+        response: The response from the GraphQL query
+
+    Returns:
+        list: A list of cleaned nodes
+    """
+    cleaned_data = []
+    edges = response.get("edges", [])
+
+    for edge in edges:
+        node = edge.get("node", {})
+        if node:
+            cleaned_data.append(node)
+
+    return cleaned_data
+
+
+def convert_to_camel_case(snake_str: str) -> str:
+    """
+    Convert snake_case string to camelCase.
+
+    Args:
+        snake_str: The snake_case string
+
+    Returns:
+        str: The camelCase string
+    """
+    snake_str = snake_str.lower()
+    components = snake_str.split("_")
+    return components[0] + "".join(x.title() for x in components[1:])
+
+
+def find_errors(response_data: dict):
+    """
+    Recursively search for a key named 'errors' in a dictionary or list of dictionaries.
+
+    Args:
+        data: The dictionary or list to search within.
+
+    Returns:
+        The value associated with the first non-empty 'errors' key found, or None if not found.
+    """
+    if isinstance(response_data, dict):
+        if "errors" in response_data and response_data["errors"]:
+            return response_data["errors"]
+        for _, value in response_data.items():
```

Comment:
```suggestion
        for value in response_data.values():
```
ref https://www.w3schools.com/python/ref_dictionary_values.asp